config/config_encryption.go

```diff
@@ -40,52 +50,52 @@ func promptForConfigEncryption() (bool, error) {
 	return common.YesOrNo(input), nil
 }
 
-// Unencrypted provides the default key provider implementation for unencrypted files
-func Unencrypted() ([]byte, error) {
-	return nil, errors.New("encryption key was requested, no key provided")
-}
-
 // PromptForConfigKey asks for configuration key
-// if initialSetup is true, the password needs to be repeated
-func PromptForConfigKey(initialSetup bool) ([]byte, error) {
-	var cryptoKey []byte
-
-	for {
-		log.Println("Please enter in your password: ")
-		pwPrompt := func(i *[]byte) error {
-			_, err := fmt.Scanln(i)
-			return err
+func PromptForConfigKey(confirmKey bool) ([]byte, error) {
+	for range 3 {
+		key, err := getSensitiveInput("Please enter encryption key: ")
+		if err != nil {
+			return nil, fmt.Errorf("%w: %w", errUserInput, err)
 		}
 
-		var p1 []byte
-		err := pwPrompt(&p1)
-		if err != nil {
-			return nil, err
+		if len(key) == 0 {
+			continue
 		}
 
-		if !initialSetup {
-			cryptoKey = p1
-			break
+		if !confirmKey {
+			return key, nil
 		}
 
-		var p2 []byte
-		log.Println("Please re-enter your password: ")
-		err = pwPrompt(&p2)
+		conf, err := getSensitiveInput("Please re-enter key: ")
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("%w: %w", errUserInput, err)
 		}
 
-		if bytes.Equal(p1, p2) {
-			cryptoKey = p1
-			break
+		if bytes.Equal(key, conf) {
+			return key, nil
+		}
+		fmt.Println("Keys did not match, please try again.")
+	}
+	return nil, fmt.Errorf("%w: %w", errUserInput, io.EOF)
+}
+
+// getSensitiveInput reads input from stdin, with echo off it stdin is a terminal
+func getSensitiveInput(prompt string) (resp []byte, err error) {
+	fmt.Print(prompt)
+	defer fmt.Println()
+	if term.IsTerminal(int(os.Stdin.Fd())) {
+		return term.ReadPassword(int(os.Stdin.Fd()))
```

Comment:
This might need a bytes.TrimRight(resp, "\r\n") as well?

Reply:
[Docs say not](https://pkg.go.dev/golang.org/x/term#ReadPassword)
> The slice returned does not include the \n.